### DISCUSSION TITLE
Suggestion: Create / use development branch

### DISCUSSION THREAD

**Discussion by ssguitarist** [Slow-Scallion4183/Adventure-Colussus#9] [General]
It may be wise to look into creating a development branch to keep unstable code away from the stable build.

The main branch should only really be used for stable/production ready code and development/feature branches should be utilised to avoid breaking code in the main branch.

You should notice the benefits of this approach as the number of contributors increases and merge conflicts start to appear.

As previously mentioned, I am still getting to understand how git / github works but believe this should be something to at least look into :)

**Comment by Slow-Scallion4183** [comment]
Yes very true! I will look into it as i have had some problems with merge conflicts (which i have now corrected).

**Comment by Slow-Scallion4183** [comment]
Ok i have created a new branch named 'development branch' and it is even with the main. Im guessing thats how i was supposed to do it, if not then please reply :D

**Reply by ssguitarist** [reply]
Looks right to me :)

**Reply by Slow-Scallion4183** [reply]
nice :) just one question: should i do all this from commandline (so typing commands into cmd) or just use the interface in the browser (in github)?

**Reply by ssguitarist** [reply]
I'm not the best person to ask as I'm still trying to work that one out myself... I tend to find myself using both for different things at present (CLI for commit/push etc, browser for merging, branching, etc). I think it's basically a "whatever suits you best", but I'm sure someone better versed with GitHub can provide more helpful feedback :)

**Reply by Slow-Scallion4183** [reply]
Thanks for the info! Just asking because i like to use github online (website) but i see programmers in youtube videos etc using CLI and i was wondering whether it would be better to use CLI vs website but that cleared it up for me! Thanks! I might just learn both and use them for different things as you said

**Reply by Slow-Scallion4183** [reply]
also what do you think we should do next?

**Reply by robotlightsyou** [reply]
I normally do PRs on the website because I want to make sure everything is documented the way I like it, but I find ```git checkout -b newbranch``` to be easier from the CLI.

**Reply by Slow-Scallion4183** [reply]
yes! i learnt it yesterday and i do think think it is a little easier from the CLI. Although, both function fine for me, so i have no complaints
